Second branch

ArtemWebior opened the pull request.

Commit:
Create lab5.py

ArtemWebior committed:
Described the task in README.md + added file "Lab5"

Also added number of task

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+        self.code = code
+        self.population = population
+        self.area = area
+        self.gdp = gdp
```
not clear name of variable

viktor-kindrat replied:
P.S. for gdp

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+        total_area = sum(country.area for country in self.countries)
+        return total_population / total_area
+
+def sort_countries_by_gdp(countries):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+        self.gdp = gdp
+        self.government_type = government_type
+
+class Land:
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+        self.name = name
+        self.countries = []
+
+    def add_country(self, country):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+    def add_country(self, country):
+        self.countries.append(country)
+
+    def calculate_population_density(self):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+def sort_countries_by_gdp(countries):
+    return sorted(countries, key=lambda country: country.gdp, reverse=True)
+
+def top_countries_by_gdp(countries, n):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+    sorted_countries = sort_countries_by_gdp(countries)
+    return sorted_countries[:n]
+
+def choose_country(countries, criteria):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+    AUTOCRACY = 3
+
+class Country:
+    def __init__(self, name, capital, code, population, area, gdp, government_type):
```
Too many arguments

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+from enum import Enum
+
+class GovernmentType(Enum):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+
+    print("============")
+
+    def criteria(country):
```
docstring missed

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,73 @@
+    REPUBLIC = 2
+    AUTOCRACY = 3
+
+class Country:
```
docstring missed

viktor-kindrat reviewed (changes requested):
There are some important fixes. Code is not clear

Commit:
Added docstrings in lab5.py

Also, changed the name for self.gdp and fixed problem with arguments in def __init__ in class Country.

viktor-kindrat commented on lab5.py:
```diff
@@ -0,0 +1,137 @@
+        government_type (GovernmentType): The type of government in the country.
+    """
+    def __init__(self, name, capital, code, gross_domestic_product, *args):
+        self.name = name
```
Too many argumentd. Recommending you changing to **kwargs

viktor-kindrat reviewed (changes requested):
Need a little more changes

Commit:
Deleted args but added kwargs to lab5.py

Also grouped things ​​into country1, country2 and country3

Commit:
Reworked Capital, Code and Gdp arguments to kwargs

Also in country1, country2 and country3 was a change according to kwargs (capital, code and gross_domestic_product)

viktor-kindrat reviewed (approved):
I approve that code is correct, clear and pass pylint test with point 10/10
